### ISSUE TITLE
Understanding how to add a Custom Runner to a .net 8 (core) project

### ISSUE THREAD

**Issue by npatel-sps** [gerardo-lijs/CrystalReportsRunner#26]
Hi,

Firstly, thank you for creating this amazing library. It's been a life saver as I have successfully used it automate exporting of 400+ Crystal reports.

However, I have a need now to be able to print the reports to specific printers. In order to do this, I have cloned the project and modified it to have an additional method called PrintToPrinter and looks a bit like this:

```csharp
 public void PrintToPrinter(Report report, string printer)
  {
      var document = ReportUtils.CreateReportDocument(report);
      document.PrintOptions.PrinterName = printer;
      // Print
      document.PrintToPrinter(nCopies: 1, collated: false, startPageN: 0, endPageN: 0);
  }
```
But I have no idea of how I can actually add these new changes to a .NET 8 worker service I'm developing. 

I have tried building the NuGet packages and add them to local NuGet packages, but I noticed that all the different projects I have tried are requiring a .NET framework dependency.

I have forked this repo and pushed the changes I made to a new branch there. If you need to see the changes, please see this commit

https://github.com/npatel-sps/CrystalReportsRunner/commit/45465631962625695da7817716ddf8c87e054db9

**Comment by gerardo-lijs** [comment]
Hi, very glad the project was helpful for you. When deciding to open source this small side project it was with the intent of helping others, so it's really nice to hear about that!

The .NET framework dependency will always be required as this is a workaround using namedpipes to communicate with the Runner which runs in .NET Framework 4.8 required by Crystal Reports.

Publising NuGet packages is tricky. If you forked and want to use it directly in your project I would avoid it directly. Anyways, I took a look at your code changes and they look fine. If you want to create a PR in this project, please do so and I can make a few changes required for it to work and then release version 1.4.3 of the library that will work for you directly.

**Comment by npatel-sps** [comment]
@gerardo-lijs Thank you for the reply and that would be great if it could be incorperated directly into library as it would save me time having to include the changes in future versions.

I will make the PR shortly. 

Thanks again.

**Comment by npatel-sps** [comment]
That's great and the single Print method is a good idea as well. No worries on the delay on the Nuget packages. 

I have a question though, if I wanted to reference the change manually within my .Net 8 Worker service, same way we do right now without the Nuget packages, is that still possible?

I understand that I can reference the core project to get all the API's. But just not sure how you can get the Runner files (.Net Framework exe and dependancies) to build alongside the .Net 8 project.

**Comment by gerardo-lijs** [comment]
Hi, will probably upload later today or tomorrow release packages.

From a .NET 8 Worker Service, if you take a look into what gets copied by the nuget packages, you will find a directory with a name like CrystalReportsRunner.13.0.35.x64 (depending which runtime you chose to use). This directory contains a net48 runner app that's independent from your application. If you compile a runner in release and copy the output to a folder like that you can deploy your own runner without the need of any nuget package.

Then, you need a reference to the LijsDev.CrystalReportsRunner.Core.dll. You can also use here your own fork and get a reference to the project directly or somehow deploy the dll in your preferred way.

I hope this small explanation helps you out. I recommend looking into Nuget packages (they are zip files basically) using NuGet package explorer or similar tool and it will help you understand what really happens behind the scene.

**Comment by gerardo-lijs** [comment]
Just published packages for version 1.4.3. Let me know if any doubt. I will close this now.

**Comment by npatel-sps** [comment]
> Hi, will probably upload later today or tomorrow release packages.
> 
> From a .NET 8 Worker Service, if you take a look into what gets copied by the nuget packages, you will find a directory with a name like CrystalReportsRunner.13.0.35.x64 (depending which runtime you chose to use). This directory contains a net48 runner app that's independent from your application. If you compile a runner in release and copy the output to a folder like that you can deploy your own runner without the need of any nuget package.
> 
> Then, you need a reference to the LijsDev.CrystalReportsRunner.Core.dll. You can also use here your own fork and get a reference to the project directly or somehow deploy the dll in your preferred way.
> 
> I hope this small explanation helps you out. I recommend looking into Nuget packages (they are zip files basically) using NuGet package explorer or similar tool and it will help you understand what really happens behind the scene.

Thank you that makes sense and I learnt something new today :)
